modules/programs/meli.nix

```
@@ -0,0 +1,222 @@
+                    description = "general shortcut configuration";
+                    example = {
+                      scroll_up = "e";
+                      scroll_down = "n";
+                    };
+                  };
+                };
+              };
+              default = { };
+              description = "Shortcut Settings";
+            };
+          };
+        };
+        default = { };
+        description = "Meli Configuration";
+      };
+    };
+    accounts.email.accounts = mkOption {
+      type = types.attrsOf (
+        types.submodule (
+          { config, ... }:
+          {
+            options.meli = {
+              enable = mkEnableOption "the meli mail client for this account";
+              mailboxes = mkOption {
+                type = with types; listOf str;
+                default = (
+                  with config.folders;
+                  [
+                    inbox
+                    sent
+                    trash
+                    drafts
+                  ]
+                );
+                example = [
+                  "INBOX"
+                  "Sent"
+                  "Trash"
+                  "Drafts"
+                ];
+                description = "Mailboxes to show in meli";
+              };
+              mailboxAliases = mkOption {
+                type = with types; attrsOf attrs;
+                default = { };
+                example = {
+                  "INBOX" = {
+                    alias = "📥 Inbox";
+                  };
+                  "Sent" = {
+                    alias = "📤 Sent";
+                  };
+                };
+                description = "Folder display name";
+              };
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = mkIf config.programs.meli.enable {
+    home.packages = [ config.programs.meli.package ];
+
+    assertions = lib.concatLists (
+      lib.attrsets.mapAttrsToList (
+        name: account:
+        lib.optional (account.meli.enable or false) {
+          assertion = account ? smtp;
+          message = "The email account '${name}' must have an 'smtp' attribute defined when enabling meli.";
+        }
+      ) config.accounts.email.accounts
+    );
+
+    # Generate meli configuration from email accounts
+    xdg.configFile."meli/config.toml".source = (pkgs.formats.toml { }).generate "meli-config" (
+      # Account Settings
+      {
+        accounts = meliAccounts;
+      }
+      //
+        # Application Settings
+        {
+          shortcuts = {
+            general = config.programs.meli.settings.shortcuts.general;
+            composing = config.programs.meli.settings.shortcuts.composing;
+            contact-list = config.programs.meli.settings.shortcuts.contact-list;
+            listing = config.programs.meli.settings.shortcuts.listing;
+            pager = config.programs.meli.settings.shortcuts.pager;
+          };
+        }
+    );
+  };
```

```suggestion
    xdg.configFile."meli/config.toml".source = (pkgs.formats.toml { }).generate "meli-config" (
      {
        accounts = meliAccounts;
      }
      // config.programs.meli.settings
    );
```

Something like this would allow more freeform support.